pkg/utils/istio/destinationrule.go

```diff
@@ -14,11 +14,58 @@ import (
 
 // DestinationRuleWithLocalityPreference returns a function setting the given attributes to a destination rule object.
 func DestinationRuleWithLocalityPreference(destinationRule *istionetworkingv1beta1.DestinationRule, labels map[string]string, destinationHost string) func() error {
-	return DestinationRuleWithLocalityPreferenceAndTLS(destinationRule, labels, destinationHost, istioapinetworkingv1beta1.ClientTLSSettings_DISABLE)
+	return DestinationRuleWithLocalityPreferenceAndTLS(destinationRule, labels, destinationHost, &istioapinetworkingv1beta1.ClientTLSSettings{Mode: istioapinetworkingv1beta1.ClientTLSSettings_DISABLE})
+}
+
+// DestinationRuleWithTLSTermination returns a function setting the given attributes to a destination rule object.
+func DestinationRuleWithTLSTermination(destinationRule *istionetworkingv1beta1.DestinationRule, labels map[string]string, destinationHost, sni, caSecret string, mode istioapinetworkingv1beta1.ClientTLSSettings_TLSmode) func() error {
+	return destinationRuleWithTrafficPolicy(
+		destinationRule,
+		labels,
+		destinationHost,
+		&istioapinetworkingv1beta1.LoadBalancerSettings{
+			LbPolicy: &istioapinetworkingv1beta1.LoadBalancerSettings_Simple{
+				Simple: istioapinetworkingv1beta1.LoadBalancerSettings_LEAST_REQUEST,
+			},
+		},
+		// OutlierDetection must be nil that simple load balancing policy takes effect.
+		nil,
```

Comment:
Do we need to specify this explicitly? What would be the default?
Does this now mean that we go for a maximum number of somewhat short-lived connections to the api servers? Are any of these connections reused or are the always recreated?

Reply:
[Istio docs](https://istio.io/latest/docs/reference/config/networking/destination-rule/#LoadBalancerSettings-SimpleLB) are clear about the default LB settings.
`No load balancing algorithm has been specified by the user. Istio will select an appropriate default.`
`LEAST_REQUEST` sounded like a reasonable choice for the beginning, so I chose it.

The default of `OutlierDetection` is `nil`. I added as parameter to `destinationRuleWithTrafficPolicy` to be able to reuse the function and make it reusable.

I can't answer the questions about short-living connections and reusing connections right now.